### PR TITLE
[9.0] hr_holidays_half_day

### PR DESCRIPTION
**Pull request by vvrossem**
New module that computes leaves in am/pm or full day.
Leaves should only by taken with calendar view as the `date_from`, `date_to` and `number_of_days_temp` field are now readonly.

This module isn't fully functional yet : we should remove all views but calendar.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Configuration for known file extensions
```
Comment (houssine78) [inline .editorconfig:1]: not sure this file should be in git

File: hr_holidays_half_day/models/hr_holidays.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+        hour, minute = floatime_to_hour_minute(hour_to)
+        utc_date_to = self._get_utc_date(date_to, hour, minute)
+        return utc_date_from, utc_date_to
+
+    def _compute_number_of_days(self, employee_id, date_from, date_to):
+        """ Returns a float equals to the timedelta between two dates given as string."""
+        hours = 0.0
+        if employee_id:
+            employee = self.env["hr.employee"].browse(employee_id)
```
Comment (houssine78) [inline hr_holidays_half_day/models/hr_holidays.py:111]: why giving the id in parameter when you have the record in hand at the function call ?

File: hr_holidays_half_day/models/hr_holidays.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            company = self.env["res.company"].browse(employee.company_id.id)
```
Comment (houssine78) [inline hr_holidays_half_day/models/hr_holidays.py]: I don't understand why you browse from the id of the company_id while you can only do 
company = employee.company_id

File: hr_holidays_half_day/models/hr_holidays.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    @api.onchange("period")
+    def onchange_period(self):
+        if self.type != "add":
+            period = self.period
+            from_dt = fields.Datetime.from_string(self.date_from)
+            to_dt = fields.Datetime.from_string(self.date_to)
+
+            company = self.env["res.company"].browse(
+                self.employee_id.company_id.id
+            )
```
Comment (houssine78) [inline hr_holidays_half_day/models/hr_holidays.py]: company = self.employee_id.company_id do the same

File: .editorconfig (hunk 1 above)
Comment (vvrossem) [reply .editorconfig:1]: It has been commited by OCA-git-bot ± a month ago : https://github.com/OCA/hr/commit/aa0f8bf37fa6941054cf10cf3c80cc467ed9f574

File: hr_holidays_half_day/views/hr_holidays_view.xml [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+            <field name="arch" type="xml">
+                <form position="attributes">
+                    <attribute name="create">false</attribute>
+                </form>
+                <field name="date_from" position="attributes">
+                    <attribute name="context">{'employee_id': employee_id, 'holiday_status_id': holiday_status_id}</attribute>
+                    <attribute name="readonly">1</attribute>
+                </field>
+                <field name="date_to" position="attributes">
+                    <attribute name="context">{'employee_id': employee_id, 'holiday_status_id': holiday_status_id}</attribute>
```
Comment (vvrossem) [inline hr_holidays_half_day/views/hr_holidays_view.xml:18]: @houssine78 the `employee_id` is given with the context

File: hr_holidays_half_day/models/hr_holidays.py [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+            diff_day = self._compute_number_of_days(
+                employee_id, date_from, date_to
+            )
+            res["value"]["number_of_days_temp"] = diff_day
+        return res
+
+    @api.multi
+    def onchange_date_to(self, date_to, date_from):
+        res = super(HrHolidays, self).onchange_date_to(date_to, date_from)
+        employee_id = self.employee_id.id or self.env.context.get(
```
Comment (vvrossem) [inline hr_holidays_half_day/models/hr_holidays.py:63]: @houssine78 I get the `employee_id` here. Same for `on_change_date_from`

File: hr_holidays_half_day/views/hr_holidays_view.xml (hunk 5 above)
Comment (houssine78) [reply hr_holidays_half_day/views/hr_holidays_view.xml:18]: not sure though that this is passed to the onchange method.

File: hr_holidays_half_day/models/hr_holidays.py (hunk 3 above)
Comment (vvrossem) [reply hr_holidays_half_day/models/hr_holidays.py]: changed in new commit

File: hr_holidays_half_day/models/hr_holidays.py (hunk 4 above)
Comment (vvrossem) [reply hr_holidays_half_day/models/hr_holidays.py]: changed in new commit

File: hr_holidays_half_day/views/hr_holidays_view.xml (hunk 5 above)
Comment (vvrossem) [reply hr_holidays_half_day/views/hr_holidays_view.xml:18]: I verified and it is passed to the onchange methods

**Comment by vvrossem** [comment]
@houssine78 ok to merge ?

**Review by houssine78 — APPROVED** [review]

**Comment by houssine78** [comment]
@vvrossem ok !

**Pull request merged**